nostr/relay.py

```diff
@@ -48,8 +56,19 @@ def connect(self, ssl_options: dict=None, proxy: dict=None):
     def close(self):
         self.ws.close()
 
+    def check_reconnect(self):
+        try:
+            self.close()
+        except:
+            pass
+        self.connected = False
+        if self.reconnect:
+            time.sleep(1)
+            self.connect(self.ssl_options)
```

Comment:
This method creates an infinite loop as long as the connection is alive. Should this be called in a thread?

Reply:
Does it? The method is only called `_on_error()` which means the connection dropped already. 

If you look into the `websocket`, it calls `websocket._app.teardown()` on error which calls `thread.join()` basically getting rid of the thread.

Reply:
`self.connect` will call `ws.run_forever`, which is where the infinite loop is created. It seems like that would block the main thread.

Reply:
> `self.connect` will call `ws.run_forever`, which is where the infinite loop is created. It seems like that would block the main thread.

Yes, but it actually doesn't run forever. It tears down the thread it's running in `_on_error`. At least that's what I gather from the code. 

I will recheck this to be sure and track the number of threads. There are lots of relays with errors these days. Will report back in a few minutes.

Reply:
Cool, thanks. No doubt there are lots of relays with errors now so we absolutely need this.

Reply:
Checked. `ws.run_forever` does not run forever. It tears down the Thread on error.

Reply:
Thanks for checking